packages/caver-core-requestmanager/caver-providers-http/src/index.js

```diff
@@ -92,17 +92,22 @@ HttpProvider.prototype.send = function(payload, callback) {
         clearTimeout(timer)
       }
       
-      if (request.readyState === 4 && request.timeout !== 1 && request.response !== null) {
+      if (request.readyState === 4 && request.timeout !== 1) {
           var result = request.responseText;
           var error = null;
 
-          try {
-              result = JSON.parse(result);
-          } catch(e) {
-              console.error(`Invalid JSON RPC response: ${JSON.stringify(request.responseText)}`)
-              error = errors.InvalidResponse(request.responseText);
+          if (request.response === null) {
+            error = errors.InvalidResponse(request.response)
+            clearTimeout(timer)
+          } else {
+            try {
+                result = JSON.parse(result);
+            } catch(e) {
+                console.error(`Invalid JSON RPC response: ${JSON.stringify(request.responseText)}`)
+                error = errors.InvalidResponse(request.responseText);
```

Comment:
Don't we need to call clearTimeout(timer) here?

Reply:
clearTimeout will be called at link 92 ! 
That timer is only for checking connection timeout, so if readyState is changed, clear that.